File: android/app/src/main/java/com/servercase/app/ui/Format.kt
package com.servercase.app.ui

import java.util.Locale

object Format {
    fun bytes(value: Double): String {
        if (value.isNaN() || value < 0) return "–"
        val units = arrayOf("B", "KB", "MB", "GB", "TB", "PB")
        var v = value
        var i = 0
        while (v >= 1024 && i < units.size - 1) { v /= 1024; i++ }
        val digits = if (v >= 100 || i == 0) 0 else 1
        return String.format(Locale.US, "%.${digits}f %s", v, units[i])
    }

    fun kb(kb: Long): String = bytes(kb * 1024.0)

    fun rate(bytesPerSec: Double?): String = if (bytesPerSec == null) "–" else "${bytes(bytesPerSec)}/s"

    fun uptime(sec: Double): String {
        if (sec <= 0) return "–"
        val total = sec.toLong()
        val d = total / 86400
        val h = (total % 86400) / 3600
        val m = (total % 3600) / 60
        return when {
            d > 0 -> "${d}d ${h}h"
            h > 0 -> "${h}h ${m}m"
            else -> "${m}m"
        }
    }
}

File: ios/ServerCase/Views/Format.swift
import Foundation
import SwiftUI

enum Format {
    static func bytes(_ value: Double) -> String {
        guard value.isFinite, value >= 0 else { return "–" }
        let units = ["B", "KB", "MB", "GB", "TB", "PB"]
        var v = value
        var i = 0
        while v >= 1024 && i < units.count - 1 { v /= 1024; i += 1 }
        let digits = (v >= 100 || i == 0) ? 0 : 1
        return String(format: "%.\(digits)f %@", v, units[i])
    }

    static func kb(_ kb: Int64) -> String { bytes(Double(kb) * 1024) }

    static func rate(_ bytesPerSec: Double?) -> String {
        guard let b = bytesPerSec else { return "–" }
        return "\(bytes(b))/s"
    }

    static func uptime(_ sec: Double) -> String {
        guard sec > 0 else { return "–" }
        let total = Int(sec)
        let d = total / 86400, h = (total % 86400) / 3600, m = (total % 3600) / 60
        if d > 0 { return "\(d)d \(h)h" }
        if h > 0 { return "\(h)h \(m)m" }
        return "\(m)m"
    }
}

enum Palette {
    static let accent = Color(red: 0.30, green: 0.55, blue: 1.0)
    static let good = Color(red: 0.25, green: 0.73, blue: 0.31)
    static let warn = Color(red: 1.0, green: 0.69, blue: 0.18)
    static let danger = Color(red: 1.0, green: 0.36, blue: 0.36)
    static let surface = Color(red: 0.086, green: 0.098, blue: 0.13)

    static func usage(_ percent: Double) -> Color {
        switch percent {
        case 90...: return danger
        case 75...: return warn
        default: return good
        }
    }
}
